rust-src/concordium_base/src/cis2_types.rs

```diff
@@ -82,6 +82,53 @@ impl TokenAmount {
     }
 }
 
+/// Full address of a token.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, Hash)]
+pub struct TokenAddress {
+    /// Address of the instance to which the token belongs.
+    contract: ContractAddress,
+    /// The id of the token inside the instance.
+    id:       TokenId,
+}
+
+impl TryFrom<&str> for TokenAddress {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let bytes = bs58::decode(value).with_check(Some(2)).into_vec()?;
+        if let Some((&2, mut source)) = bytes.split_first() {
+            let index = leb128::read::unsigned(&mut source)?;
```

Comment:
Is the LEB128 really needed here? We will probably end up implementing this in multiple languages, so unless it provides good value, I think we should keep it simple.

Reply:
It makes for much shorter token IDs, which I think is a good tradeoff.

Reply:
fair enough